File: Problems & Solutions/520. Detect Capital/02 January 2022/Approach 1 - Character by Character/Detect Capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        int n = word.length();

        boolean match1 = true, match2 = true, match3 = true;

        for (int i = 0; i < n; i++) {
            if (!Character.isUpperCase(word.charAt(i))) {
                match1 = false;
                break;
            }
        }
        if (match1) {
            return true;
        }

        for (int i = 0; i < n; i++) {
            if (!Character.isLowerCase(word.charAt(i))) {
                match2 = false;
                break;
            }
        }
        if (match2) {
            return true;
        }

        if (!Character.isUpperCase(word.charAt(0))) {
            match3 = false;
        }
        if (match3) {
            for (int i = 1; i < n; i++) {
                if (!Character.isLowerCase(word.charAt(i))) {
                    match3 = false;
                    break;
                }
            }
        }
        if (match3) {
            return true;
        }

        return false;
    }
}

File: Problems & Solutions/520. Detect Capital/02 January 2022/Approach 1 - Character by Character/Detect Capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        n = len(word)
            
        match1, match2, match3 = True, True, True

        for i in range(n):
            if not word[i].isupper():
                match1 = False
                break
        if match1:
            return True

        for i in range(n):
            if word[i].isupper():
                match2 = False
                break
        if match2:
            return True

        if not word[0].isupper():
            match3 = False
        if match3:
            for i in range(1, n):
                if word[i].isupper():
                    match3 = False
        if match3:
            return True

        return False